.github/workflows/goreleaser.yml

```diff
@@ -35,6 +29,15 @@ jobs:
         with:
           fetch-tags: 1
           fetch-depth: 1
+          
+      # Set environment variables required by GoReleaser
+      - name: Set build environment variables
+        run: |
+          echo "GIT_STATE=$(if git diff-index --quiet HEAD --; then echo 'clean'; else echo 'dirty'; fi)" >> $GITHUB_ENV
```

Comment:
Git has the ability to automatically tack on `-dirty` or `-broken` to the tag/commit if the tree has working changes or issues using the `--dirty` and `--broken` flags to `git describe`. Is having a separate field preferred over this?

Reply:
At this level, we don't have control over how {{ .Version }} is generated before it is inserted into the build context.